sui_programmability/framework/sources/Governance/SuiSystem.move

```diff
@@ -153,21 +159,94 @@ module Sui::SuiSystem {
         )
     }
 
+    public(script) fun request_add_delegation(
+        self: &mut SuiSystemState,
+        delegate_stake: Coin<SUI>,
+        validator_address: address,
+        ctx: &mut TxContext,
+    ) {
+        let amount = Coin::value(&delegate_stake);
```

Comment:
How do we make sure that amounts are valid (ie non zero)? Is there like a protection or it is okay to pass `Coin::zero()` inside?